platform/server/plugins/types.ts

```diff
@@ -1,7 +1,78 @@
-import { KibanaPluginFeatures, As } from '../../types';
+import { Observable } from 'rxjs';
+
+import { As, Schema } from '../../types';
+import * as schemaLib from '../../lib/schema';
+import { ConfigWithSchema } from '../../config';
+import { Router, RouterOptions } from '../http';
+import { KibanaConfig } from '../kibana';
+import { ElasticsearchService, ElasticsearchConfigs } from '../elasticsearch';
+import { LoggerFactory } from '../../logger';
 
 export type PluginName = string & As<'PluginName'>;
 
+export interface KibanaPluginFeatures {
+  /**
+   * Plugin-scoped logger
+   */
+  logger: LoggerFactory;
+
+  /**
+   * Core Kibana utilities
+   */
+  util: {
+    schema: Schema;
+  };
+
+  /**
+   * Core Elasticsearch functionality
+   */
+  elasticsearch: {
+    service: ElasticsearchService;
+    config$: Observable<ElasticsearchConfigs>;
+  };
+  kibana: {
```

Comment:
__nit:__ JSDoc for `kibana` property maybe, just for the sake of consistency (and full stop at the end of every sentence in JSDoc if you don't mind this rule)?